src/generators/nodes/Window.ts

```diff
@@ -170,8 +178,10 @@ export default class Window extends Node {
 
 			block.builders.init.addBlock(deindent`
 				#component.observe("${bindings.scrollX || bindings.scrollY}", function(${isX ? 'x' : 'y'}) {
-					if (${lock}) return;
+					${lock} = true;
+					clearTimeout(${timeout});
 					window.scrollTo(${isX ? 'x, window.scrollY' : 'window.scrollX, y'});
+					${timeout} = setTimeout(function() { ${lock} = false; }, 100);
```

Comment:
I'm not entirely clear on how this stuff gets optimized - but would this create a new anonymous function on every observed change of `x`/`y`? Do we want that function to be declared out at the same level as the lock variable is declared?

Reply:
That's a good point — have updated it to play safe